lib/rpm.rb

```diff
@@ -41,11 +41,19 @@ def self.transaction(root = '/')
   # @param [String] name Name of the macro
   # @return [String] value of macro +name+
   def self.[](name)
-    val = ''
-    buffer = ::FFI::MemoryPointer.new(:pointer, 1024)
-    buffer.write_string("%{#{name}}")
-    ret = RPM::C.expandMacros(nil, nil, buffer, 1024)
-    buffer.read_string
+    if C::rpm_version_code >= ((4 << 16) + (14 << 8) + (0 << 0))
+      obuf = ::FFI::MemoryPointer.new(:pointer)
+      sbuf = FFI::MemoryPointer.from_string("%{#{name}}")
+      ret = RPM::C.rpmExpandMacros(nil, sbuf, obuf, 0)
+      val = obuf.get_pointer(0)
+      val.read_string
+    else
+      val = ''
```

Comment:
This val is not needed (Removed in #11)

Reply:
Yes I still need to rebase this on your changes, done